packages/checkbox/src/react/index.tsx

```diff
@@ -65,113 +64,101 @@ const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
       () => (ref.current as unknown) as HTMLInputElement
     )
 
-    const square = React.createRef<HTMLDivElement>()
-
     React.useEffect(
       function updateIndetermiateAttr() {
         if (!ref.current) return
-        ref.current.indeterminate = props.indeterminate || false
+        ref.current.indeterminate = indeterminate || false
       },
-      [props.indeterminate]
+      [indeterminate]
     )
 
-    const handleKeyDown = (evt: React.KeyboardEvent<HTMLLabelElement>) => {
-      if (evt.key === 'Enter' || evt.key === ' ') handleClick(evt)
-    }
-
-    const handleClick = (
+    const toggle = (
       evt:
-        | React.KeyboardEvent<HTMLLabelElement>
         | React.MouseEvent<HTMLLabelElement>
+        | React.KeyboardEvent<HTMLLabelElement>
     ) => {
+      if (onCheck && isFunction(onCheck)) onCheck(evt, !checked, value, name)
+    }
+
+    const handleClick: React.MouseEventHandler<HTMLLabelElement> = evt => {
+      if (disabled) return
+
       evt.preventDefault()
       evt.stopPropagation()
+      evt.persist()
+
+      toggle(evt)
+    }
 
-      const { onCheck } = props
-      if (typeof onCheck === 'function') {
-        onCheck(evt, !props.checked, props.value, props.name)
-      }
+    const handleKeyDown: React.KeyboardEventHandler<HTMLLabelElement> = evt => {
+      if (disabled) return
 
-      if (square.current) square.current.focus()
```

Comment:
i removed this feature to be more inline with the native browser interaction. 🤷

Reply:
What *was* this feature?  Look like it's for keeping focus on the checkbox after clicking.  That seems like a browser-native thing to do.  I'll checkout and try out...

Reply:
it shows the *halo* focus indicator when a checkbox is clicked with a mouse. this was different that the native implementation so i changed it. gifs attached

native behaviour
![Nov-04-2020 15-10-37](https://user-images.githubusercontent.com/226356/98173276-10363f00-1eb0-11eb-9ace-b9026f5ce0cc.gif)

new behavior that matches native

![Nov-04-2020 15-13-14](https://user-images.githubusercontent.com/226356/98173433-5a1f2500-1eb0-11eb-943b-a8c4c71e4d5a.gif)